src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -0,0 +1,1344 @@
+                item_meta["domain_id"][ts_index] = output_meta["domain_id"]
+            else:
+                item_meta["index"][ts_index] = output_meta["index"]
+
+    app_version_info = {}
+    for index, raw_result_file in result_files.items():
+        app_version_info[index] = raw_result_file[META_GROUP_NAME].attrs.get(
+            "application_version"
+        )
+
+        skip = update_helper(
+            profiles_helper,
+            index,
+            global_profiles_metadata,
+            profiles_base_ts_to_time_set_keys,
+        )
+        if skip is not skip_sentinel:
+            read_profiles_statistics(index)
+
+        skip = update_helper(
+            trends_helper,
+            index,
+            global_trends_metadata,
+            trends_base_ts_to_time_set_keys,
+        )
+        if skip is not skip_sentinel:
+            read_trends_statistics(index)
+
+    # Put collected statistics into metadata.
+    for output_key, metadata_item in profiles_helper.meta.items():
+        profile_statistics = global_profiles_statistics[output_key]
+        metadata_item["time_set_key"] = profiles_to_time_set_key[output_key]
+        metadata_item["global_max"] = profile_statistics["global_max"]
+        metadata_item["global_min"] = profile_statistics["global_min"]
+
+    for output_key, metadata_item in trends_helper.meta.items():
+        trend_statistics = trends_statistics[output_key]
+        metadata_item["time_set_key"] = trends_to_time_set_key[output_key]
+        metadata_item["max"] = float(trend_statistics["max"])
+        metadata_item["min"] = float(trend_statistics["min"])
+
+    return _MergedMetadataWithStatistics(
+        app_version_info=app_version_info,
+        profile_key_to_metadata=profiles_helper.meta,
+        trend_key_to_metadata=trends_helper.meta,
+        time_set_keys=list(time_set_ranges.keys()),
+    )
+
+
+def _read_global_metadata(
+    result_files: Dict[int, h5py.File],
+) -> Tuple[BaseTimeStepIndexToMetaList, BaseTimeStepIndexToMetaList]:
+    """
+    :return:
+        - a dict mapping base time steps to a list of profile metadata items;
+        - a dict mapping base time steps to a list of trend metadata items;
+    """
+    import json
+
+    def update_global_metadata(
+        result_file: h5py.File,
+        metadata_collection: BaseTimeStepIndexToMetaList,
+        base_ts: int,
+        output_type: str,
+    ) -> None:
+        """
+        Read a metadata file and add it to a metadata collection.
+
+        :param metadata_collection:
+            This collection is changed. The keys are the "base time step index"
+            and the values are the metadata from the file (the duplicates have
+            been removed and the original index recorded).
+        """
+        meta = json.loads(result_file[META_GROUP_NAME].attrs[output_type])
+        # # Remove duplicates from the result. Save the original index.
```

Comment:
Commented out code?

Reply:
I will remove this.
It was already commented in alfasim for a long time =)